test/images/agnhost/openidmetadata/openidmetadata.go

```diff
@@ -162,3 +173,27 @@ func withOAuth2Client(context.Context) (context.Context, error) {
 		})
 	return ctx, nil
 }
+
+// DNS can be available sometime after the container starts due to the way
+// networking is set up for Windows nodes with dockershim as the container runtime.
+// In this case, we should make sure we are able to resolve the issuer before
+// invoking oidc.NewProvider.
+// See https://github.com/kubernetes/kubernetes/issues/99470 for more details.
+func ensureWindowsDNSAvailability(issuer string) error {
+	log.Println("Ensuring Windows DNS availability")
+
+	u, err := url.Parse(issuer)
+	if err != nil {
+		return err
+	}
+
+	return wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
```

Comment:
Is a 5 second timeout going to be enough?

Reply:
I ran the test case multiple times and it usually takes around 1 second to return and 5 seconds should be more than enough.